### PR TITLE
APP-4896: Implement GetAuthApplication API

### PR DESCRIPTION
**Pull request by jr22**
APP-4896

Implementing GetAuthApplication API that will eventually result in CLI command with format: 
```
viam auth-app get --org-id=c6215428-1b73-41c3-b44a-56db0631c8f1 --application-id=6847e7a9-d7bd-4603-9655-bdc24370342a 
{
	"application_id": "6847e7a9-d7bd-4603-9655-bdc24370342a",
	"application_name": "julias app",
	"secret": "supersupersecretsecret"
	"origin_uris": ["https://test.com", "https://test2.com" ],
	"redirect_uris": ["https://redirect.com", "https://redirect2.com" ],
	"logout_uri": "https://logout.com",
}
```

### CODE REVIEW

File: proto/viam/app/v1/end_user.proto [hunk 1]
```diff
@@ -57,3 +60,17 @@ message UpdateAuthApplicationResponse {
...
+
+message GetAuthApplicationRequest {
+  string org_id = 1;
+  string application_id = 2;
+}
+
+message GetAuthApplicationResponse {
+  string application_id = 1;
+  string application_name = 2;
+  string secret = 3;
```
Comment (ohEmily) [inline proto/viam/app/v1/end_user.proto:72]: [minor] maybe we rename to `client_secret` to match FusionAuth's `application.oauthConfiguration.clientSecret` from their [API](https://fusionauth.io/docs/apis/applications)? also makes it a little easier to grep for this string.

**Review by ohEmily — APPROVED** [review]
Thank you! Added a small comment.

File: proto/viam/app/v1/end_user.proto (hunk 1 above)
Comment (jr22) [reply proto/viam/app/v1/end_user.proto:72]: yeah makes sense to me. i can also rename the `secret` field in the `RegisterAuthApplication` request [here](https://github.com/viamrobotics/api/pull/506/files#diff-a9b71a46ce21ecb89e78164be49f980caadad4344c1116b519e8fc41dafd5d7eR47) to client_secret then?

**Comment by jr22** [comment]
workflow is broken for api changes -> fixing [here](https://github.com/viamrobotics/api/pull/507)

**Comment by ohEmily** [comment]
Weird that changing this broke the linter. Seems like a linter bug. To unblock your PR you could squash the changes.

```
/ Test For Lint and Breaking Changes
Field "3" with name "client_secret" on message "RegisterAuthApplicationResponse" changed option "json_name" from "secret" to "clientSecret".
 Check failure on line 47 in proto/viam/app/v1/end_user.proto

/ Test For Lint and Breaking Changes
Field "3" on message "RegisterAuthApplicationResponse" changed name from "secret" to "client_secret".
```

**Comment by ohEmily** [comment]
> > Weird that changing this broke the linter suite. Seems like a linter bug. To unblock your PR you could squash the changes.
> > ```
> > / Test For Lint and Breaking Changes
> > Field "3" with name "client_secret" on message "RegisterAuthApplicationResponse" changed option "json_name" from "secret" to "clientSecret".
> >  Check failure on line 47 in proto/viam/app/v1/end_user.proto
> > 
> > / Test For Lint and Breaking Changes
> > Field "3" on message "RegisterAuthApplicationResponse" changed name from "secret" to "client_secret".
> > ```
> 
> Its a breaking change for me to change the RegisterAuthResponse field from secret -> client_secret so i think this failure is expected?

Oh what I meant is that it seems dumb to consider it a "breaking" change when it's comparing to an earlier version of the same PR. IMO it should only compare to main, not to the current branch.

**Comment by jr22** [comment]
> > > Weird that changing this broke the linter suite. Seems like a linter bug. To unblock your PR you could squash the changes.
> > > ```
> > > / Test For Lint and Breaking Changes
> > > Field "3" with name "client_secret" on message "RegisterAuthApplicationResponse" changed option "json_name" from "secret" to "clientSecret".
> > >  Check failure on line 47 in proto/viam/app/v1/end_user.proto
> > > 
> > > / Test For Lint and Breaking Changes
> > > Field "3" on message "RegisterAuthApplicationResponse" changed name from "secret" to "client_secret".
> > > ```
> > 
> > 
> > Its a breaking change for me to change the RegisterAuthResponse field from secret -> client_secret so i think this failure is expected?
> 
> Oh what I meant is that it seems dumb to consider it a "breaking" change when it's comparing to an earlier version of the same PR. IMO it should only compare to main, not to the current branch.

It is comparing to main though. On main the "RegisterAuthApplicationResponse" has "secret" not "client_secret". Maybe im not understanding your q?

**Pull request merged**